astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -1358,7 +1359,7 @@ def test_data_at_range_limit(parallel, fast_reader, guess):
         fast_reader["parallel"] = parallel
         if fast_reader.get("use_fast_converter"):
             rtol = 1.0e-15
-        elif np.iinfo(np.int_).dtype == np.dtype(np.int32):
+        elif sys.maxsize < 2**32:
             # On 32bit the standard C parser (strtod) returns strings for these
             pytest.xfail("C parser cannot handle float64 on 32bit systems")
```

Comment:
Another case where we don't even seem to need the `xfail` but are instead missing a warning. Strangely the failure is
> >       with pytest.warns() as warning_lines:
E       Failed: DID NOT WARN. No warnings of type (<class 'Warning'>,) were emitted.
E       The list of emitted warnings is: [].

while the test below in L1410 actually expects `len(warning_lines) in (0, 1)` – looks like that part has already been broken for 0 warnings before, probably with some pytest update.

Reply:
But there isn't any new pytest release this past weekend.

Reply:
No, this would have broken already when changing this from `pytest.warns(None)` to `pytest.warns()` in https://github.com/astropy/astropy/pull/12823#issuecomment-1030254653 – this overlooked that there _was_ a case where 0 warnings were expected, my reading of this is that only 32bit systems do not raise any warning here and thus the error was never discovered while the `xfail` was in effect.

Reply:
Is the Windows job really 32-bit though? The log says `MSC v.1929 64 bit (AMD64)` even though it is `win32`. 🤯

Reply:
It is passing all the reads up to e.g. https://github.com/astropy/astropy/actions/runs/5997690440/job/16264575442#step:10:1868
just like any 64-bit system, so is not failing the precision test, just not emitting an expected warning in 2 cases.